### DISCUSSION TITLE
[10월 1주차] 나머지가 1이되는 수 찾기

### DISCUSSION THREAD

**Discussion by z1nun** [z1nun/ASTERISK-#28] [알고리즘 문제풀이]
<h6 class="guide-section-title" style="box-sizing: border-box; padding: 0px; margin: 0px 0px 1rem; color: white; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-weight: 700; font-stretch: inherit; font-size: 14px; line-height: 1.5; font-family: Inter, NotoSansKR, &quot;Helvetica Neue&quot;, Helvetica, Arial, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Noto Color Emoji&quot;; -webkit-font-smoothing: antialiased; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(38, 55, 71); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; user-select: auto;">문제 설명</h6><div class="markdown solarized-dark" style="box-sizing: border-box; padding: 0px; margin: 0px; color: rgb(178, 192, 204); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-weight: 400; font-stretch: inherit; font-size: 16px; line-height: inherit; font-family: Inter, NotoSansKR, &quot;Helvetica Neue&quot;, Helvetica, Arial, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Noto Color Emoji&quot;; background-color: rgb(38, 55, 71); word-break: normal; overflow-wrap: break-word; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; user-select: auto;"><p style="box-sizing: border-box; padding: 0px; margin: 0px; color: inherit; font-style: inherit; font-variant: inherit; font-weight: 400; font-stretch: inherit; font-size: inherit; line-height: 1.8; font-family: Inter, NotoSansKR, &quot;Helvetica Neue&quot;, Helvetica, Arial, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Noto Color Emoji&quot;; letter-spacing: -0.1px; user-select: auto; overflow-wrap: break-word;">자연수<span> </span><code style="box-sizing: border-box; padding: 2px 6px; margin: 0px 2px 0px 0px; color: rgb(205, 215, 224); font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 14px; line-height: inherit; font-family: Hack, Menlo, Monaco, Consolas, &quot;Courier New&quot;, monospace, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif; overflow-wrap: break-word; border: 1px solid rgb(23, 35, 52); background-color: rgb(32, 43, 61); border-radius: 0.25rem; user-select: auto;">n</code>이 매개변수로 주어집니다.<span> </span><code style="box-sizing: border-box; padding: 2px 6px; margin: 0px 2px 0px 0px; color: rgb(205, 215, 224); font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 14px; line-height: inherit; font-family: Hack, Menlo, Monaco, Consolas, &quot;Courier New&quot;, monospace, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif; overflow-wrap: break-word; border: 1px solid rgb(23, 35, 52); background-color: rgb(32, 43, 61); border-radius: 0.25rem; user-select: auto;">n</code>을<span> </span><code style="box-sizing: border-box; padding: 2px 6px; margin: 0px 2px 0px 0px; color: rgb(205, 215, 224); font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 14px; line-height: inherit; font-family: Hack, Menlo, Monaco, Consolas, &quot;Courier New&quot;, monospace, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif; overflow-wrap: break-word; border: 1px solid rgb(23, 35, 52); background-color: rgb(32, 43, 61); border-radius: 0.25rem; user-select: auto;">x</code>로 나눈 나머지가 1이 되도록 하는 가장 작은 자연수<span> </span><code style="box-sizing: border-box; padding: 2px 6px; margin: 0px 2px 0px 0px; color: rgb(205, 215, 224); font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 14px; line-height: inherit; font-family: Hack, Menlo, Monaco, Consolas, &quot;Courier New&quot;, monospace, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif; overflow-wrap: break-word; border: 1px solid rgb(23, 35, 52); background-color: rgb(32, 43, 61); border-radius: 0.25rem; user-select: auto;">x</code>를 return 하도록 solution 함수를 완성해주세요. 답이 항상 존재함은 증명될 수 있습니다.</p><hr style="box-sizing: content-box; padding: 0px; margin: 1.5rem 0px; color: inherit; font: inherit; height: 0px; overflow: visible; border-width: 0.0625rem 0px 0px; border-right-style: initial; border-bottom-style: initial; border-left-style: initial; border-color: rgb(23, 35, 52); border-image: initial; border-top-style: solid; user-select: auto;"><h5 style="box-sizing: border-box; padding: 0px; margin: 1.5rem 0px 0px; color: white; font-style: inherit; font-variant: inherit; font-weight: 400; font-stretch: inherit; font-size: 15px; line-height: 1.5; font-family: Inter, NotoSansKR, &quot;Helvetica Neue&quot;, Helvetica, Arial, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Noto Color Emoji&quot;; letter-spacing: -0.009em; user-select: auto;">제한사항</h5><ul style="box-sizing: border-box; padding: 0px 0px 0px 2.5rem; margin: 0.75rem 0px 0px; color: inherit; font: inherit; user-select: auto;"><li style="box-sizing: border-box; padding: 0px; margin: 0px; color: inherit; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 16px; line-height: 1.6; font-family: inherit; letter-spacing: -0.009em; list-style: inherit; user-select: auto;">3 ≤<span> </span><code style="box-sizing: border-box; padding: 2px 6px; margin: 0px 2px 0px 0px; color: rgb(205, 215, 224); font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 14px; line-height: inherit; font-family: Hack, Menlo, Monaco, Consolas, &quot;Courier New&quot;, monospace, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif; overflow-wrap: break-word; border: 1px solid rgb(23, 35, 52); background-color: rgb(32, 43, 61); border-radius: 0.25rem; user-select: auto;">n</code><span> </span>≤ 1,000,000</li></ul><hr style="box-sizing: content-box; padding: 0px; margin: 1.5rem 0px; color: inherit; font: inherit; height: 0px; overflow: visible; border-width: 0.0625rem 0px 0px; border-right-style: initial; border-bottom-style: initial; border-left-style: initial; border-color: rgb(23, 35, 52); border-image: initial; border-top-style: solid; user-select: auto;"><h5 style="box-sizing: border-box; padding: 0px; margin: 1.5rem 0px 0px; color: white; font-style: inherit; font-variant: inherit; font-weight: 400; font-stretch: inherit; font-size: 15px; line-height: 1.5; font-family: Inter, NotoSansKR, &quot;Helvetica Neue&quot;, Helvetica, Arial, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Noto Color Emoji&quot;; letter-spacing: -0.009em; user-select: auto;">입출력 예</h5>

n | result
-- | --
10 | 3
12 | 11

<hr style="box-sizing: content-box; padding: 0px; margin: 1.5rem 0px; color: inherit; font: inherit; height: 0px; overflow: visible; border-width: 0.0625rem 0px 0px; border-right-style: initial; border-bottom-style: initial; border-left-style: initial; border-color: rgb(23, 35, 52); border-image: initial; border-top-style: solid; user-select: auto;"><h5 style="box-sizing: border-box; padding: 0px; margin: 1.5rem 0px 0px; color: white; font-style: inherit; font-variant: inherit; font-weight: 400; font-stretch: inherit; font-size: 15px; line-height: 1.5; font-family: Inter, NotoSansKR, &quot;Helvetica Neue&quot;, Helvetica, Arial, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Noto Color Emoji&quot;; letter-spacing: -0.009em; user-select: auto;">입출력 예 설명</h5><p style="box-sizing: border-box; padding: 0px; margin: 0.75rem 0px 0px; color: inherit; font-style: inherit; font-variant: inherit; font-weight: 400; font-stretch: inherit; font-size: inherit; line-height: 1.8; font-family: Inter, NotoSansKR, &quot;Helvetica Neue&quot;, Helvetica, Arial, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Noto Color Emoji&quot;; letter-spacing: -0.1px; user-select: auto; overflow-wrap: break-word;"><strong style="box-sizing: border-box; padding: 0px; margin: 0px; color: inherit; font-style: inherit; font-variant: inherit; font-weight: bolder; font-stretch: inherit; font-size: inherit; line-height: inherit; font-family: inherit; -webkit-font-smoothing: antialiased; user-select: auto;">입출력 예 #1</strong></p><ul style="box-sizing: border-box; padding: 0px 0px 0px 2.5rem; margin: 1rem 0px 0px; color: inherit; font: inherit; user-select: auto;"><li style="box-sizing: border-box; padding: 0px; margin: 0px; color: inherit; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 16px; line-height: 1.6; font-family: inherit; letter-spacing: -0.009em; list-style: inherit; user-select: auto;">10을 3으로 나눈 나머지가 1이고, 3보다 작은 자연수 중에서 문제의 조건을 만족하는 수가 없으므로, 3을 return 해야 합니다.</li></ul><p style="box-sizing: border-box; padding: 0px; margin: 0.75rem 0px 0px; color: inherit; font-style: inherit; font-variant: inherit; font-weight: 400; font-stretch: inherit; font-size: inherit; line-height: 1.8; font-family: Inter, NotoSansKR, &quot;Helvetica Neue&quot;, Helvetica, Arial, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Noto Color Emoji&quot;; letter-spacing: -0.1px; user-select: auto; overflow-wrap: break-word;"><strong style="box-sizing: border-box; padding: 0px; margin: 0px; color: inherit; font-style: inherit; font-variant: inherit; font-weight: bolder; font-stretch: inherit; font-size: inherit; line-height: inherit; font-family: inherit; -webkit-font-smoothing: antialiased; user-select: auto;">입출력 예 #2</strong></p><ul style="box-sizing: border-box; padding: 0px 0px 0px 2.5rem; margin: 1rem 0px 0px; color: inherit; font: inherit; user-select: auto;"><li style="box-sizing: border-box; padding: 0px; margin: 0px; color: inherit; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 16px; line-height: 1.6; font-family: inherit; letter-spacing: -0.009em; list-style: inherit; user-select: auto;">12를 11로 나눈 나머지가 1이고, 11보다 작은 자연수 중에서 문제의 조건을 만족하는 수가 없으므로, 11을 return 해야 합니다.</li></ul></div>

https://school.programmers.co.kr/learn/courses/30/lessons/87389

**Comment by sweetyr928** [comment]
```js
function solution(n) {
    for(let i=2;i<n;i++){
        if(n%i===1){
            return i;
        }
    }
}
```

**Comment by 2Zerozero** [comment]
```
function solution(n) {
    let answer = 0;
    
    // n % x === 1 이 되게하는 가장 작은 자연수
    // x 를 찾는 방법은 무엇?
    for(let x = 0; x < n; x++) {
        if(n % x === 1) {
            answer = x;
            break;
        }
    }
    return answer;
}
```

**Comment by iltae** [comment]
```js
function solution(n) {
    // 1로 나누면 나머지가 다 0이기에 1은 생략 2부터 시작
    // 나머지가 1이 되는 x를 찾고 break으로 탈출
    for( let x = 2; x < n; x++) {
        if ( n % x === 1) {
            return x;
            break;
        }
    }
}
```

**Comment by rhino-ty** [comment]
```js
function solution(n) {
  for (let i = 3; i < n; i++) {
    if (n % i === 1) {
      return i;
    }
  }
}
// 소수에서 했던 것처럼 2 제외 시키기
function solution(n) {
  if (n % 2 === 1) {
    return 2;
  } else
    for (let i = 3; i < n; i++) {
      if (n % i === 1) {
        return i;
      }
    }
}
```

**Comment by fairytale779** [comment]
```js
function solution(n) {
    let x = 1;
    while (x<n){
        if (n % x === 1) return x
        else x++
    } 
}

**Comment by FordangIT** [comment]
```js
function solution(n) {
    for(let i=2; i<n; i++){
        if(n%i ===1)
            return i
    }
}
```

**Comment by FordangIT** [comment]
1)
```js
 function solution(n) {
    for(let i=2; i<n; i++){
        if(n%i ===1)
            return i
    }
}
```
2) 
```js
function solution(n, x = 1) {    
    while (x++) {
        if (n % x === 1) {
            return x;
        }
    }    
}
```
3) 
```js
function solution(n, x = 0) {
   return n % x === 1 ? x : solution(n, x + 1);
// 테스트 한 개가 런타임 에러가 나온다. }
```

**Comment by z1nun** [comment]
```JS
const solution = function(n) {
    for (let i=0; i<n; i++){
        if (n%i == 1){
            return i
        }
    }
}
```
